receiver.go

```diff
@@ -117,7 +61,7 @@ func (r *Receiver) Receive(ctx context.Context) (*Message, error) {
 		// and keeps the behavior the same as before the unsettled messages tracking was introduced
 		defer r.link.DeleteUnsettled(&msg)
```

Comment:
this breaks the modesecond. that's the whole point of the HandleMessage api. you can't delete the message as soon as you return it to the caller, it's not disposed yet, so it's not settled, and the credit should not be issued.

Reply:
Yeah, I think we need to do a manual creditor check here too and opt out if it's enabled. 

Ie, tracking unsettled messages is the "autocredit management" path.

Reply:
It still needs to move though.  We can't remove the message from the unsettled map until the message has been disposed.

Reply:
I know it's not straight forward but are you planning to build the autocredit management on top of the manual credit, instead of aside of it?

Reply:
Opened https://github.com/Azure/go-amqp/issues/73 to track consolidation.